core/src/com/unciv/ui/utils/extensions/FormattingExtensions.kt

```diff
@@ -19,6 +19,9 @@ fun Float.toPercent() = 1 + this/100
 /** Convert a [resource name][this] into "Consumes [amount] $resource" string (untranslated) */
 fun String.getConsumesAmountString(amount: Int) = "Consumes [$amount] [$this]"
 
+/** Convert a [resource name][this] into "Need [amount] more $resource" string (untranslated) */
+fun String.getNeedMoreAmountString(amount: Int) = "Need [$amount] more [$this]"
```

Comment:
If you make a new string, also add it to the `template.properties` file, so that it can be translated into other langauges

Reply:
got it